ddtrace/tracer/option.go

```diff
@@ -96,6 +96,9 @@ type config struct {
 	// propagator propagates span context cross-process
 	propagator Propagator
 
+	// injector injects span context cross-process
+	injector ExperimentalInjector
```

Comment:
Hmm I'm wondering if there's a way around introducing another value here. (I'm just thinking out loud here since I'm not 100% sure it would actually be better)
If this `ExperimentalInjector` could be considered a "kind" of propagator. We could treat Propagator kind of like how go sql approaches the `Conn` interface, where the `WithOptions` bit is an enhanced interface that we use if it's implemented otherwise either fallback to the normal Inject (or perhaps do a no-op). hmmm Typing all this out makes me think it wouldn't actually be better.... I think maybe the issue I'm having is trying to distinguish between this `injector` and the `propagator` above. What makes them different and how can we better express that in the code (or comments) so as to not confuse down the line